File: backend/model/blog.model.js
import mongoose from "mongoose"

const BlogSchema = new mongoose.Schema(
  {
    title: String,
    summary: String,
    content: String,
    coverImage: String,
    author: String,
    blogId: { type: String, required: true, unique: true },
    date: { type: String },
  },
  { timestamps: true }
)

const BlogModel = mongoose.model("Blog", BlogSchema)

export default BlogModel

File: backend_v2/src/main/java/com/d3vilGhost/blogify/model/BlogModel.java
package com.d3vilGhost.blogify.model;

import com.mongodb.lang.NonNull;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.bson.types.ObjectId;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.index.Indexed;
import org.springframework.data.mongodb.core.mapping.Document;

@Data
@Document(collection = "blog")
@NoArgsConstructor
public class BlogModel {
    @Id
    private ObjectId id;
    @Indexed(unique = true)
    @NonNull
    private String blogId;
    @NonNull
    private String title;
    @NonNull
    private String summary;
    @NonNull
    private String content;
    @NonNull
    private String coverImage;
    @NonNull
    private String author;
    @NonNull
    private String date;
}
